### DISCUSSION TITLE
Undefine result on DatabaseGpt.Web

### DISCUSSION THREAD

**Discussion by djtyambao** [marcominerva/DatabaseGPT#77] [Q&A] [answered]
Hi Marco,

I am experiencing some challenges with my deployment on the server environment (IIS web). Specifically, I am encountering an "Undefined" error. While troubleshooting using the console version, I was able to view the results without any issues. Additionally, the web version I tested in Visual Studio worked perfectly. The error only occurs when the application is deployed on the server. I have attached screenshots from my troubleshooting process for reference. Is there a possibility that OpenAI is being blocked when deployed on the server? I would appreciate your help with this issue.

Thank you
Jake C.

![WebServerDeployError](https://github.com/user-attachments/assets/1c09f3bf-cea8-4584-8164-437a9ba1b237)

![ServerdeployConsole](https://github.com/user-attachments/assets/9de30f29-5757-4370-86ce-017a091c825d)
![VS2022_result](https://github.com/user-attachments/assets/871eb4c2-6181-4fe7-a393-5c1377cfa46f)

**Comment by marcominerva** [comment]
Try to check in the Windows Event Viewer to see if you find some details about the problem.

**Reply by djtyambao** [reply]
I checked the Event Viewer, but I didn't find any errors from the application. The IIS user has full control of the folder.  All the necessary SDK's are installed.

**Reply by marcominerva** [reply]
Have you checked also in the **Application** section under **Windows Logs**?

**Reply by djtyambao** [reply]
Yes I did.

**Reply by djtyambao** [reply]
Here's the latest warning that I got. Do I need to update the Nugets?
Category: Microsoft.AspNetCore.DataProtection.KeyManagement.XmlKeyManager
EventId: 35
No XML encryptor configured. Key {d2479b32-26d0-42f5-b983-3dff24ee4803} may be persisted to storage in unencrypted form.

**Reply by marcominerva** [reply]
It's only a warning, it isn't related to your issue. The Browser Developer Toolbar contains some Logs?

**Reply by djtyambao** [reply]
![errorbrowser](https://github.com/user-attachments/assets/a233c274-055f-4dec-b17a-ce9a28d17650)

**Reply by marcominerva** [reply]
At what local address have you deployed the site?

**Reply by djtyambao** [reply]
I am using a demo server -> https://demoOS-jphlcu3/ChatDB

**Reply by djtyambao** [reply]
Can the application only be deployed on an Azure server?

**Reply by marcominerva** [reply, marked as answer]
The default configuration expects that the Application will be deployed on a top level domain. You're using a second level site, so you need to modify line 9 of `app.js` file removing the initial slash in the `fetch` call.

**Reply by djtyambao** [reply]
I changed the app.js from fetch('/api/chat/ask' to fetch('api/chat/ask'). Still the same error.

**Reply by marcominerva** [reply]
Have you forced a full page reload using CTRL+F5?

**Reply by djtyambao** [reply]
Hello, 
Now I'm receiving an error after modifying the app.js file and clearing the cache. I verified the connection string that  I obtained from the VStudio app settings. 
![image](https://github.com/user-attachments/assets/7af959b3-c183-4bd8-b460-b81a39f973f8)

**Reply by marcominerva** [reply]
The erorr message clearly tells that the connection string you have set in the `appsettings.json` file contains invalid credentials.

**Reply by djtyambao** [reply]
I will check on that. Thank you for your patience.

**Reply by djtyambao** [reply]
It is functioning properly; I need to set some SQL credentials, such as the network service, and also include the domain\machinename$. Thank you again much appreciated.
